test/PublishCommand.js

```diff
@@ -995,4 +992,48 @@ describe("PublishCommand", () => {
       });
     });
   });
+
+  /** =========================================================================
+   * VERSION LIFECYCLE SCRIPTS
+   * ======================================================================= */
+
+  describe("lifecycle scripts", () => {
+    let testDir;
+
+    const scripts = ["preversion", "version", "postversion"];
+
+    beforeEach(async () => {
+      testDir = await initFixture("PublishCommand/lifecycle");
+    });
+
+    it("should call version lifecycle scripts for a package", async () => {
+      await run(testDir)();
+      scripts.forEach(script => {
+        expect(NpmUtilities.runScriptInDir).toHaveBeenCalledWith(
+          script,
+          [],
+          path.resolve(testDir, "packages", "package-2"),
+          jasmine.any(Function)
+        );
+      });
+    });
+
+    it("should not call version lifecycle scripts for a package missing them", async () => {
+      await run(testDir)();
+      scripts.forEach(script => {
+        expect(NpmUtilities.runScriptInDir).not.toHaveBeenCalledWith(
+          script,
+          [],
+          path.resolve(testDir, "packages", "package-2"),
+          jasmine.any(Function)
```

Comment:
Pretty sure `expect.any(Function)` should work here? Or is this Jest 19 (not upgraded yet for reasons) biting us again?